foursight_core/fs_connection.py

```diff
@@ -46,18 +46,20 @@ def __init__(self, fs_environ, fs_environ_info, test=False, use_es=True, host=No
         self.ff_env = fs_environ_info['ff_env']
         self.ff_es = fs_environ_info['es']
         self.ff_bucket = fs_environ_info['bucket']
+        self.redis = None
+        self.redis_url = None
         try:
             if 'redis' in fs_environ_info:
-                self.redis = RedisBase(create_redis_client(url=fs_environ_info['redis']))
+                self.redis_url = fs_environ_info['redis']
+                self.redis = RedisBase(create_redis_client(url=self.redis_url))
             elif 'REDIS_HOST' in os.environ:  # temporary patch in until env config is fully sorted - Will
-                self.redis = RedisBase(create_redis_client(url=os.environ['REDIS_HOST']))
-            else:
-                self.redis = None
+                self.redis_url = os.environ['REDIS_HOST']
+                self.redis = RedisBase(create_redis_client(url=self.redis_url))
```

Comment:
The setup line should be factored out and occur only once after the `if` statement. It doesn't need to be repeated in each clause unless the initialization is different in each case, which it doesn't look like it is. Also want to notice if neither setup is done. Something like (untested):
```
            if 'redis' in fs_environ_info:
                self.redis_url = fs_environ_info['redis']
            elif 'REDIS_HOST' in os.environ:  # temporary patch in until env config is fully sorted - Will
                self.redis_url = os.environ['REDIS_HOST']
            if self.redis_url:
                self.redis = RedisBase(create_redis_client(url=self.redis_url)
            else:
                PRINT("Redis URL was not specified in any way, so no RedisBase is being created.")
```

Reply:
Yes good points, thanks.